components/validator/challenger.go

```diff
@@ -65,191 +93,413 @@ func NewChallenger(ctx context.Context, cfg Config, l log.Logger) (*Challenger,
+			c.log.Info("validating output", "l2BlockNumber", l2BlockNumber, "outputRoot", expected.Hex())
+		case <-c.done:
+			return
+		}
+	}
 }
 
-func (c *Challenger) DetermineChallengeTx() (*types.Transaction, error) {
-	// Check for a challenge in progress.
-	isInProgress, err := c.IsChallengeInProgress()
+// subscribeChallengeCreated subscribes the ChallengeCreated event from Colosseum contract and handle challenge.
+func (c *Challenger) subscribeChallengeCreated() {
+	defer c.wg.Done()
+
+	// ChallengeCreated events
+	event := c.colosseumABI.Events["ChallengeCreated"]
+	query := goEth.FilterQuery{
+		Addresses: []common.Address{c.cfg.ColosseumAddr},
+		Topics: [][]common.Hash{
+			{event.ID},
+		},
+	}
+
+	logs := make(chan goTypes.Log)
+	sub, err := c.l1Client.SubscribeFilterLogs(context.Background(), query, logs)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get challenge in progress: %w", err)
+		c.log.Error("failed to subscribe to logs", "err", err)
+		return
 	}
 
-	if isInProgress {
-		isRelated, err := c.IsRelatedChallenge()
-		if err != nil {
-			return nil, fmt.Errorf("unable to check relationship with challenge: %w", err)
-		}
+	for {
+		select {
+		case err := <-sub.Err():
+			c.log.Error("failed to subscribe to logs", "err", err)
+			return
+		case vLog := <-logs:
+			if len(vLog.Topics) == 0 {
+				c.log.Error("logs from ChallengeCreated event are empty")
+				break
+			}
 
-		if !isRelated {
-			c.log.Info("another challenge is in progress")
-			return nil, nil
+			outputIndex := new(big.Int).SetBytes(vLog.Topics[3][:])
+			// when challenge created, handle it
+			if outputIndex.Sign() == 1 {
+				c.wg.Add(1)
+				go c.handleChallenge(outputIndex)
+			}
+		case <-c.done:
+			return
 		}
+	}
+}
 
-		status, err := c.GetStatusInProgress()
-		if err != nil {
-			return nil, fmt.Errorf("unable to get challenge status: %w", err)
-		}
+// handleChallenge handles challenge according to its status and role.
+func (c *Challenger) handleChallenge(outputIndex *big.Int) {
+	defer c.wg.Done()
 
-		if !c.cfg.OutputSubmitterDisabled {
-			switch status {
-			case chal.StatusAsserterTurn:
-				return c.Bisect()
-			case chal.StatusChallengerTimeout:
-				challengeId, err := c.GetChallengeInProgress()
-				if err != nil {
-					return nil, err
-				}
+	// get challenge
+	challenge, err := c.GetChallenge(outputIndex)
+	if err != nil {
+		c.log.Error("failed to get challenge", "err", err, "outputIndex", outputIndex)
+		return
+	}
+
+	isAsserter := challenge.Asserter == c.cfg.TxManager.From()
+	isChallenger := challenge.Challenger == c.cfg.TxManager.From()
+
+	// not related challenge
+	if !isAsserter && !isChallenger {
+		c.log.Error("not related challenge", "outputIndex", outputIndex)
+		return
+	}
 
-				// TODO(pangssu): Is it necessary to submit challengerTimeout transaction?
-				c.log.Info("challenger timed out", "challengeId", challengeId)
-				return nil, nil
+	ticker := time.NewTicker(c.cfg.PollInterval)
```

Comment:
I think you should separate this PollInterval from the PollInterval that is used by a validator. This is because output root is submitted almost every 1 hour and you should poll rarely. But for the challenge, you should poll quite a lot because it should react to the opponent.